### PR TITLE
support switch load different environment config file.

### PR DESCRIPTION
**Pull request by Ashuaidehao**
This feature could let developers switch different environment(mainnet/testnet) config more easily.
Just set the "NEO_ENV" environment variable value before run it. 
Please don't hesitate to let me know if there is any problem.

### CODE REVIEW

**Comment by erikzhang** [comment]
I think changing file name is easier then setting environment variable.

File: neo/Helper.cs [hunk 1]
```diff
@@ -307,5 +308,20 @@ internal static long WeightedAverage<T>(this IEnumerable<T> source, Func<T, long
...
+
+        /// <summary>
+        /// load configuration with different Environment Variable
+        /// </summary>
+        /// <param name="config"></param>
+        /// <returns></returns>
+        public static IConfigurationRoot LoadConfig(string config)
+        {
+            var env = Environment.GetEnvironmentVariable("NEO_ENV");
+            var configFile = string.IsNullOrWhiteSpace(env) ? $"{config}.json" : $"{config}.{env}.json";
```
Comment (shargon) [inline neo/Helper.cs]: If is not defined, you will expect `config..json`

**Comment by shargon** [comment]
In this cases, you don't need to define the env variable, and use the regular mechanism

File: neo/Helper.cs [hunk 2]
```diff
@@ -307,5 +308,20 @@ internal static long WeightedAverage<T>(this IEnumerable<T> source, Func<T, long
...
+
+
+        /// <summary>
+        /// load configuration with different Environment Variable
+        /// </summary>
+        /// <param name="config"></param>
+        /// <returns></returns>
+        public static IConfigurationRoot LoadConfig(string config)
+        {
+            var fileEnv = Environment.GetEnvironmentVariable($"NEO_{config.ToUpperInvariant()}_FILE");
```
Comment (erikzhang) [inline neo/Helper.cs]: If we read from `NEO_XXX_FILE`, this method is not reusable. We can move it from `Helper.cs` to `ProtocolSettings.cs`. And mark it as `private`.

**Comment by Ashuaidehao** [comment]
> I think changing file name is easier then setting environment variable.
This feature is basically useful for docker developers , which hardly change file name in container but easily set environment variable .

File: neo/Helper.cs (hunk 2 above)
Comment (Ashuaidehao) [reply neo/Helper.cs]: I think this method can be used for all the "config" file loading(e.g. protocal.json, config.json, all other plugin.json). Batch change the config files by just setting one environment variable value.

Comment (erikzhang) [reply neo/Helper.cs]: Then we should have only one environment variable for switching network.

**Comment by shargon** [comment]
Shall I revert my changes?

File: neo/Helper.cs [hunk 3]
```diff
@@ -317,8 +317,9 @@ internal static long WeightedAverage<T>(this IEnumerable<T> source, Func<T, long
         /// <returns></returns>
         public static IConfigurationRoot LoadConfig(string config)
         {
-            var fileEnv = Environment.GetEnvironmentVariable($"NEO_{config.ToUpperInvariant()}_FILE");
-            var configFile = string.IsNullOrWhiteSpace(fileEnv) ? $"{config}.json" : fileEnv;
+            var env = Environment.GetEnvironmentVariable("NEO_ENV");
+            var configFile = string.IsNullOrWhiteSpace(env) ? $"{config}.json" : $"{config}.{env}.json";
+            Console.WriteLine($"Config loading:{configFile}");
```
Comment (erikzhang) [inline neo/Helper.cs]: Should remove this line or use `Log()`.

File: neo/Helper.cs [hunk 4]
```diff
@@ -307,5 +308,21 @@ internal static long WeightedAverage<T>(this IEnumerable<T> source, Func<T, long
...
+
+
+        /// <summary>
+        /// load configuration with different Environment Variable
+        /// </summary>
+        /// <param name="config"></param>
+        /// <returns></returns>
+        public static IConfigurationRoot LoadConfig(string config)
+        {
+            var env = Environment.GetEnvironmentVariable("NEO_ENV");
```
Comment (erikzhang) [inline neo/Helper.cs]:
```suggestion
            var env = Environment.GetEnvironmentVariable("NEO_NETWORK");
```

File: neo/Helper.cs [hunk 5]
```diff
@@ -307,5 +308,21 @@ internal static long WeightedAverage<T>(this IEnumerable<T> source, Func<T, long
...
             }
         }
+
+
+        /// <summary>
+        /// load configuration with different Environment Variable
+        /// </summary>
+        /// <param name="config"></param>
+        /// <returns></returns>
+        public static IConfigurationRoot LoadConfig(string config)
```
Comment (erikzhang) [inline neo/Helper.cs]: I still think this can't be reused. Because the config files are not always at the root application directory.

Comment (Ashuaidehao) [reply neo/Helper.cs]: As my test, we can pass the parameter "config" like "Plugins/config" which can surpport different directory.

Comment (erikzhang) [reply neo/Helper.cs]: Most of the plugin has only one config file. In this case, if they use this method, they will try to read from a file that does not exist.

Comment (Ashuaidehao) [reply neo/Helper.cs]: You are right . As far as I know, the plugin config files rarely change along with the network, so these scenarios wouldn't use this method. The most common usage is loading "config.json" and "protocal.json".  Mark it public so neo-cli can call it to load "config.json".

**Comment by erikzhang** [comment]
Can you port it to 3.0? We are not going to improve on 2.x.

**Comment by Ashuaidehao** [comment]
Sure ,here(https://github.com/neo-project/neo/pull/851) is the PR for 3.0

**Comment by vncoelho** [comment]
I agree with @erikzhang, even considering that these things are minor are useful, if we start to modify NEO 2.x we are going to have to test and focus on a different line of reasoning.

**Pull request closed without merging**